examples/bearriver/src/FRP/BearRiver.hs

```diff
@@ -170,11 +188,26 @@ occasionally tAvg b
   timeDelta :: Monad m => SF m a DTime
   timeDelta = arrM_ ask
 
+-- | Initialization operator (cf. Lustre/Lucid Synchrone).
+--
+-- The output at time zero is the first argument, and from
+-- that point on it behaves like the signal function passed as
+-- second argument.
 (-->) :: Monad m => b -> SF m a b -> SF m a b
 b0 --> sf = MSF $ \a -> do
   (_, ct) <- unMSF sf a
   return (b0, ct)
 
+-- | Input initialization operator.
+--
+-- The input at time zero is the first argument, and from
+-- that point on it behaves like the signal function passed as
+-- second argument.
+(>--) :: Monad m => a -> SF m a b -> SF m a b
+a0 >-- sf = MSF $ \_ -> do
+  (b, ct) <- unMSF sf a0
+  return (b, ct)
```

Comment:
This could be factorized:
```haskell
replaceOnce :: a -> MSF a a -> MSF a a
replaceOnce x sf = MSF $ \_ -> unMSF x

(-->) :: Monad m => b -> SF m a b -> SF m a b 
b0 --> sf = sf >>> replaceOnce b0

(>--) :: Monad m => a -> SF m a b -> SF m a b 
a0 >-- sf = replaceOnce a0 >>> sf
```

The question is: can one write `replaceOnce` without using `unMSF`? It's possible, using switching :)

Reply:
@turion You've implemented a new exception-handling mechanism. What's the current way of writing `\x -> dSwitch (constant (x, Event)) (\_ -> id)` using that API?

Reply:
Awesome :)
Shouldn't ```replaceOnce``` look like: 
```haskell
replaceOnce :: Monad m => a -> MSF m a a
replaceOnce x = MSF $ \_ -> return (x, arr id)
```

Reply:
And here without using the internal MSF constructor:
```haskell
replaceOnce :: Monad m => a -> SF m a a
replaceOnce a = dSwitch (arr $ const (a, Event ())) (const $ arr id)
```

Reply:
```haskell
replaceOnce :: Monad m => a -> SF m a a
replaceOnce a = dSwitch (constant (a, Event ())) (const $ arr id)
```

?

Reply:
Yeah obviously ;)